tools/apigee-envoy-quickstart/extension/README.md

```diff
@@ -0,0 +1,145 @@
+    export CLUSTER_LOCATION=<gke-cluster-region>
+    export APIGEE_ORG=<apigee-org-tied-to-apigee-envoy-adapter>
+    ```
+
+1. **Set up local authentication to your project.**
+    ```bash
+    gcloud config set project $GKE_PROJECT_ID
+    gcloud auth application-default login --no-launch-browser
+
+    export TOKEN=$(gcloud auth print-access-token)
+    ```
+
+1. **Set up login profile for your cluster.**
+    ```bash
+    gcloud container clusters get-credentials "$CLUSTER_NAME" \
+     --zone "$CLUSTER_LOCATION" \
+     --project "$GKE_PROJECT_ID"
+     export CLUSTER_CTX="gke_${GKE_PROJECT_ID}_${CLUSTER_LOCATION}_${CLUSTER_NAME}"
+    ```
+
+1. ### Set the namespace hosting istio-ingressgateway
+    ```bash
+    export $ISTIO_GATEWAY_NS=<namespace-hosting-ingressgateway>
+    ```  
+
+1. ### Execute the below if istio-ingressgateway is missing in the istio enabled cluster
+    ```bash
+    git clone https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages.git \
+    /tmp/asm
+
+    kubectl create namespace $ISTIO_GATEWAY_NS
+
+    kubectl label namespace $ISTIO_GATEWAY_NS istio-injection=enabled istio.io/rev-
+
+    kubectl apply -n $ISTIO_GATEWAY_NS \
+        -f /tmp/asm/samples/gateways/istio-ingressgateway
+
+    kubectl rollout restart deployment -n $ISTIO_GATEWAY_NS
+
+    kubectl get svc istio-ingressgateway -n $ISTIO_GATEWAY_NS
+    ```
+
+1. **Overwrite the service details that has to be exposed outside the mesh, if different from the service (httpbin application) deployed in the [starter kit](https://github.com/apigee/devrel/tree/main/tools/apigee-envoy-quickstart#envoy-with-apigee-adapter-as-containers-within-kubernetes-platform)**
+    ```bash
+    export TARGET_HOST="httpbin.org"
+    export TARGET_SERVICE_NAME="httpbin"
+    export TARGET_SERVICE_NAMESPACE="apigee"
+    ```
+
+1. **Fetch the Loadbalancer details exposing the ingress gatway**
+    ```bash
+    export INGRESS_HOST=$(kubectl -n $ISTIO_GATEWAY_NS get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    export INGRESS_PORT=$(kubectl -n $ISTIO_GATEWAY_NS get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].port}')
+    export SECURE_INGRESS_PORT=$(kubectl -n $ISTIO_GATEWAY_NS get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].port}')
+    export TCP_INGRESS_PORT=$(kubectl -n $ISTIO_GATEWAY_NS get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="tcp")].port}')
+    ```
+
+1. **Setup Gateway component**
+    ```bash
+   cat <<EOF > /tmp/httpbin-gateway.yaml              
+   apiVersion: networking.istio.io/v1alpha3
+   kind: Gateway
+   metadata:
+     name: httpbin-gateway
+   spec:
+     selector:
+       istio: ingressgateway
+     servers:
+       - port:
+           number: 80
+           name: http
+           protocol: HTTP
+         hosts:
+           - "$TARGET_HOST"
+   EOF
+
+   kubectl apply -n $ISTIO_GATEWAY_NS -f /tmp/httpbin-gateway.yaml
```

Comment:
Can we pipe the template output above directly into kubectl? Fewer files dangling in /tmp

```
cat <<EOF | kubectl apply -f -
...
EOF
```

Reply:
fixed.